app/controllers/v0/health_care_applications_controller.rb

```diff
@@ -80,11 +80,20 @@ def facilities
     private
 
     def active_facilities(lighthouse_facilities)
-      active_ids = StdInstitutionFacility.active.pluck(:station_number).compact
+      active_ids = cached_ves_facility_ids
+
+      if active_ids.empty?
+        HCA::StdInstitutionImportJob.new.perform
+        active_ids = cached_ves_facility_ids
+      end
 
       lighthouse_facilities.select { |facility| active_ids.include?(facility.unique_id) }
     end
```

Comment:
I just have one nit suggestion

```ruby
def active_facilities(lighthouse_facilities)
  lighthouse_facilities.select { |facility| active_ves_facility_ids.include?(facility.unique_id) }
end

def active_ves_facility_ids
  @active_ves_facility_ids ||= begin
    HCA::StdInstitutionImportJob.new.perform if cached_ves_facility_ids.empty?
    cached_ves_facility_ids
  end
end
```

Reply:
I think this reads a bit clearer 👍

Reply:
Hm, agreed on the restructuring to push the conditional run deeper, but I hoped to keep it to one DB query when possible. Maybe more of a guard clause pattern would help:
```
def active_facilities(lighthouse_facilities)
  lighthouse_facilities.select { |facility| active_ves_facility_ids.include?(facility.unique_id) }
end

def active_ves_facility_ids
  ids = cached_ves_facility_ids
  return ids if ids.any?

  HCA::StdInstitutionImportJob.new.perform 

  cached_ves_facility_ids
end
```

Also, I'm not sure the memoization would work? I thought each request to a Controller runs in its own instance, so `@active_ves_facility_ids` will be reset to `nil` every time one comes in, and we won't save anything by storing it. Since `[]` is truthy in Ruby, we can't even use `||=` to conditionally assign it the way we could in JS.

Reply:
IMHO cleaner to assign to a local variable outside the loop rather than have a multiline `begin` block for the assignment. Also, Rubocop objects to an early-`return` inside the assignment, which makes the whole thing more cumbersome still.